### PR TITLE
Regularization for BiGAN, KLpq, GAN and WGAN inferences

### PR DESCRIPTION
**Pull request by siddharth-agrawal**
Adds regularization in the loss functions for BiGAN, KLpq, GAN and WGAN inferences. This makes further progress towards fixing #529.

### CODE REVIEW

**Comment by siddharth-agrawal** [comment]
@dustinvtran I'm not confident about the change I made in `klpq.py`. Specifically about the fact that I'm subtracting from the loss instead of adding to it. I did this because the gradients are being taken for `-loss`.

**Comment by dustinvtran** [comment]
What you did is correct for both variational and model parameters in `klpq`.

File: edward/inferences/wgan_inference.py [hunk 1]
```diff
@@ -90,10 +93,14 @@ def build_loss_and_gradients(self, var_list):
                                      list(range(1, gradients.shape.ndims))))
       penalty = self.penalty * tf.reduce_mean(tf.square(slopes - 1.0))
 
+    reg_terms_d = tf.losses.get_regularization_losses(scope="Disc")
+    reg_terms_all = tf.losses.get_regularization_losses()
```
Comment (dustinvtran) [inline edward/inferences/wgan_inference.py]: Any reason this one  and GAnInference uses all but BiGAN's uses `scope="Gen"`?

Comment (siddharth-agrawal) [reply edward/inferences/wgan_inference.py]: I went by how variable lists are created in the respective files. For GAN and WGAN, `scope="Gen"` is not used for creating `var_list` so I assumed it wasn't there.

Comment (dustinvtran) [reply edward/inferences/wgan_inference.py]: That makes sense. Merging now. All looks great.
